python/gui/MaaS/static/maas/js/components/createDatasetForm.js

```diff
@@ -0,0 +1,42 @@
+class CreateDatasetForm {
+    constructor(parentDivId) {
+        this.parentDivId = parentDivId;
+        this.formElementId = this.parentDivId + "-form";
+        this.formContentDivId = this.formElementId + "-div-universal-inputs";
+        this.dynamicVarsDivId = this.formElementId + "-div-dynamic-inputs";
+
+    }
+
+    updateFormatChange(selection) {
+        let dy_div = document.getElementById(this.dynamicVarsDivId);
+        //let initial_length = dy_div.childNodes.length;
+        dy_div.childNodes.forEach(child => child.remove());
```

Comment:
Related to #198. See [this](https://github.com/NOAA-OWP/DMOD/issues/198#issuecomment-1289327285) comment specifically. This should be changed to the following:

```javascript
// remove all dynamic input child elements
while (dy_div.firstChild) {
  dy_div.removeChild(dy_div.firstChild);
}
```